### PR TITLE
Rework immucore

### PR DESCRIPTION
**Pull request by Itxaka**

### CODE REVIEW

File: pkg/op/op.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+	"github.com/containerd/containerd/mount"
+	"github.com/kairos-io/immucore/internal/constants"
+	internalUtils "github.com/kairos-io/immucore/internal/utils"
+	"github.com/kairos-io/immucore/pkg/state"
+	"github.com/rs/zerolog"
+	"time"
+)
+
+// MountOP creates and executes a mount operation.
+func MountOP(what, where, t string, options []string, timeout time.Duration, s *state.State) func(context.Context) error {
```
Comment (mudler) [inline pkg/op/op.go:16]: maybe better to call this file `mount.go`? as we are already in `pkg/op`

**Comment by jimmykarily** [comment]
It's mostly moving things around but I can't tell if there is any new functionality added. I guess the best way to test this is to run the test suite with it, right?

**Comment by Itxaka** [comment]
> It's mostly moving things around but I can't tell if there is any new functionality added. I guess the best way to test this is to run the test suite with it, right?

nothing changed in terms of functionality, just moving and renaming things to make it more clear whats what and groups things better

**Review by jimmykarily — APPROVED** [review]

**Comment by Itxaka** [comment]
Run the kairos uki tests against this branch and it passed all tests: https://github.com/kairos-io/kairos/actions/runs/8357383176?pr=2372

so merging!

**Pull request merged**